### PR TITLE
Store each relation child in its own key

### PR DESCRIPTION
**Pull request by someone235**

### CODE REVIEW

File: consensus/src/model/stores/children.rs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    pub fn insert_batch(&self, batch: &mut WriteBatch, parent: Hash, child: Hash) -> Result<(), StoreError> {
+        self.access.write(BatchDbWriter::new(batch), parent, child)?;
+        Ok(())
+    }
+}
+
+impl ChildrenStoreReader for DbChildrenStore {
+    fn get(&self, parent: Hash) -> StoreResult<ReadLock<BlockHashSet>> {
+        self.access.read(parent) // TODO: Pass read lock
```
Comment (michaelsutton) [inline consensus/src/model/stores/children.rs:104]: Is this TODO up-to-date?

File: consensus/src/pipeline/header_processor/processor.rs [hunk 2]
```diff
@@ -124,7 +124,7 @@ pub struct HeaderProcessor {
     db: Arc<DB>,
 
     // Stores
-    pub(super) relations_stores: Arc<RwLock<Vec<DbRelationsStore>>>,
+    pub(super) relations_stores: Arc<RwLock<Vec<DbRelationsStore>>>, // TODO: Remove RwLock
```
Comment (michaelsutton) [inline consensus/src/pipeline/header_processor/processor.rs:127]: what about this TODO?

File: consensus/src/pipeline/header_processor/processor.rs [hunk 3]
```diff
@@ -417,7 +417,7 @@ impl HeaderProcessor {
 
         let mut relations_write = self.relations_stores.write();
         ctx.known_parents.into_iter().enumerate().for_each(|(level, parents_by_level)| {
-            relations_write[level].insert_batch(&mut batch, header.hash, parents_by_level).unwrap();
+            relations_write[level].insert_batch(&mut batch, header.hash, parents_by_level.clone()).unwrap();
```
Comment (michaelsutton) [inline consensus/src/pipeline/header_processor/processor.rs:420]: why the new `clone()` here?

File: consensus/src/consensus/mod.rs [hunk 4]
```diff
@@ -760,7 +760,12 @@ impl ConsensusApi for Consensus {
     }
 
     fn get_block_children(&self, hash: Hash) -> Option<Arc<Vec<Hash>>> {
-        self.services.relations_service.get_children(hash).unwrap_option()
+        self.services
+            .relations_service
+            .get_children(hash)
+            .unwrap_option()
+            .map(|children| Arc::new(children.read().iter().copied().collect_vec()))
+        // TODO: No need for Arc
```
Comment (michaelsutton) [inline consensus/src/consensus/mod.rs:768]: what about this TODO?

File: consensus/src/consensus/factory.rs [hunk 5]
```diff
@@ -54,6 +54,20 @@ pub struct MultiConsensusMetadata {
     version: u32,
 }
 
+const LATEST_DB_VERSION: u32 = 1;
+impl Default for MultiConsensusMetadata {
+    fn default() -> Self {
+        Self {
+            current_consensus_key: Default::default(),
```
[the remark this answers is not in the document]
Comment (someone235) [reply consensus/src/consensus/factory.rs:61]: This will cause an infinite loop

**Review by michaelsutton — APPROVED** [review]

**Review by michaelsutton — APPROVED** [review]

**Pull request merged**